site/layouts/partials/sidebar.html

```diff
@@ -44,35 +20,46 @@
-                {{ end }}
-                {{ if $subexpanded }}
-                <ul class="sub-menu" style="display: block">
-                {{ else }}
-                <ul class="sub-menu">
-                {{ end }}
-                {{ range .Pages.ByWeight }}
-                    {{ if eq $.Title .Title }}
-                    <a class="item active" href="{{ .Permalink }}">{{ .Title }}</a>
+                <span class="item {{ if $active }}active{{ end }}">
+                    <i class="sub-menu__toggle fa {{$.Scratch.Get "icon"}}" aria-hidden="true" style="{{$.Scratch.Get "iconstyle"}}"></i>
+                    {{ if $active }}
+                        <a class="active" href="#"> {{ .Title }}</a>
                     {{ else }}
-                    <a class="item" href="{{ .Permalink }}">{{ .Title }}</a>
+                        <a href="{{ .Permalink }}"> {{ .Title }}</a>
+                    {{ end }}
+                </span>
+                <ul class="sub-menu" {{ if $expanded }}style="display: block"{{ end }}>
+                    {{ range .Pages.ByWeight }}
+                        {{ $subactive := (eq $.Title .Title) }}
+                        {{ $subexpanded := or (eq .Title $.Params.subsection) $subactive }}
+
+                        {{ $.Scratch.Set "subicon" "fa-plus-square-o" }}
+                        {{ if $subexpanded }}
+                            {{ $.Scratch.Set "subicon" "fa-minus-square-o" }}
+                        {{ end }}
+
+                        {{if eq (len (.Pages)) 0}}
+                            {{ $.Scratch.Set "iconstyle" "display: none" }}
+                        {{ else }}
+                            {{ $.Scratch.Set "iconstyle" "" }}
+                        {{ end }}
+
+                        <span class="item {{ if $subactive }}active{{ end }}">
+                            <i class="sub-menu__toggle fa {{$.Scratch.Get "subicon"}}" aria-hidden="true" style="{{$.Scratch.Get "iconstyle"}}"></i>
+                            {{ if $subactive }}
+                                <a class="active" href="{{ .Permalink }}"> {{ .Title }}</a></span>
+                            {{ else }}
+                                <a href="{{ .Permalink }}"> {{ .Title }}</a></span>
+                            {{ end }}
+                        </span>
+                        <ul class="sub-menu" {{ if $subexpanded }}style="display: block"{{ end }}>
+                            {{ range .Pages.ByWeight }}
+                                <a class="item {{ if eq $.Title .Title }}active{{ end }}" href="{{ .Permalink }}">{{ .Title }}</a>
+                            {{ end }}
+                        </ul>
                     {{ end }}
-                {{ end }}
                 </ul>
             {{ end }}
-            {{ range .Pages.ByWeight }}
```

Comment:
Essentially, this block was duplicating elements. By just changing the above block to always iterate on `.Pages.ByWeight`, we get the desired behaviour for free, including obviating the need to make overview a folder just to mark it as a section.